### PR TITLE
Fix SDK integration tests on Windows

### PR DESCRIPTION
**Pull request by cocreature**
I haven’t yet managed to get the maven stuff to work inside of the Bazel test (it does work outside) but I think having the rest of the integration tests run on Windows is far better than having no integration tests on Windows so for now I simply disabled them on Windows.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/integration-tests/src/Main.hs [hunk 1]
```diff
@@ -27,31 +29,39 @@ import DA.Bazel.Runfiles
 import DamlHelper
 import SdkVersion
 
+-- | Since we run in bash and not in cmd.exe "daml" won’t look for "daml.cmd"
+-- so we use "daml.cmd" directly.
+damlName :: String
+damlName
```
Comment (neil-da) [inline daml-assistant/integration-tests/src/Main.hs]: Why not just move to using the Shell form of process? In truth, that's what most users will be doing, so its the most accurate test and no need to have daml.cmd vs daml.

File: daml-assistant/integration-tests/src/Main.hs [hunk 2]
```diff
@@ -160,7 +172,12 @@ quickstartTests quickstartDir mvnDir = testGroup "quickstart"
                   (socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr))
                   close
                   (\s -> connect s (addrAddress addr))
-    , testCase "mvn compile" $
+    ] <>
+    -- The mvn tests seem to fail on Windows for some reason so for now we disable them.
```
Comment (neil-da) [inline daml-assistant/integration-tests/src/Main.hs]: Can we get a ticket to have this fixed at some point?

File: daml-assistant/integration-tests/src/Main.hs [hunk 3]
```diff
@@ -189,10 +206,21 @@ quickstartTests quickstartDir mvnDir = testGroup "quickstart"
     where
         mvnRepoFlag = "-Dmaven.repo.local=" <> mvnDir
 
+-- | Bazel tests are run in a bash environment with cmd.exe not in PATH. This results in ShellCommand
```
Comment (neil-da) [inline daml-assistant/integration-tests/src/Main.hs]: Is that not fixable?

File: daml-assistant/integration-tests/src/Main.hs (hunk 1 above)
Comment (neil-da) [reply daml-assistant/integration-tests/src/Main.hs]: Ah, I found adjustCP below...

Comment (cocreature) [reply daml-assistant/integration-tests/src/Main.hs]: Moved it after `adjustCP` and pointed to the docs of `adjustCP`.

File: daml-assistant/integration-tests/src/Main.hs (hunk 2 above)
Comment (cocreature) [reply daml-assistant/integration-tests/src/Main.hs]: Opened a ticket and linked to it from this comment.

File: daml-assistant/integration-tests/src/Main.hs (hunk 3 above)
Comment (cocreature) [reply daml-assistant/integration-tests/src/Main.hs]: Haven’t investigated it too deeply but so far I haven’t found a nice solution.

**Pull request merged**